src/lib/common/types/content-entry.ts

```diff
@@ -31,3 +38,125 @@ export enum ContentEntryStatus {
   Published = "published",
   Archived = "archived",
 }
+
+export type ContentEntry = {
+  archivedAt?: Maybe<Scalars["Time"]>;
+  blueprintBranch?: Maybe<Scalars["String"]>;
+  blueprintId?: Maybe<Scalars["String"]>;
+  blueprintVariant?: Maybe<Scalars["String"]>;
+  createdAt?: Maybe<Scalars["Time"]>;
+  createdByUserId?: Maybe<Scalars["String"]>;
+  documentId?: Maybe<Scalars["String"]>;
+  environmentId?: Maybe<Scalars["String"]>;
+  firstPublishedAt?: Maybe<Scalars["Time"]>;
+  lastUpdatedByUserId?: Maybe<Scalars["String"]>;
+  previewImageUrl?: Maybe<Scalars["String"]>;
+  projectId?: Maybe<Scalars["String"]>;
+  publishedAt?: Maybe<Scalars["Time"]>;
+  status?: Maybe<ContentEntryStatus>;
+  title?: Maybe<Scalars["String"]>;
+  unpublishedAt?: Maybe<Scalars["Time"]>;
+  updatedAt?: Maybe<Scalars["Time"]>;
+  fields?: Maybe<Array<ContentEntryField>>;
+};
+
+export type ContentEntryField = {
+  id?: Maybe<Scalars["String"]>;
+  blueprintFieldId?: Maybe<Scalars["String"]>;
+  createdAt?: Maybe<Scalars["Time"]>;
+  data?: Maybe<Scalars["Any"]>;
+  documentFieldLocaleId?: Maybe<Scalars["String"]>;
+  lastUpdatedByUserId?: Maybe<Scalars["String"]>;
+  type?: Maybe<ContentEntryContentTypeFieldType>;
+  updatedAt?: Maybe<Scalars["Time"]>;
+};
+
+export type ContentEntryFieldData = {
+  valueString?: string | null;
+  valueBool?: boolean | null;
+  valueKeywords?: string | null;
+  valueDate?: string | null;
+  valueNumber?: number | null;
+  valueObjects?: string | null;
+};
+
+export const processDataForEntryField = (data: Maybe<Scalars["Any"]>): ContentEntryFieldData => {
+  if (data === null || data === undefined) {
+    return {};
+  } else if (typeof data === "string") {
+    try {
+      const parsedDate = Date.parse(data);
+      if (!isNaN(parsedDate)) {
+        return {
+          valueDate: new Date(parsedDate).toISOString(),
+          valueString: undefined,
+          valueBool: undefined,
+          valueNumber: undefined,
+          valueKeywords: undefined,
+          valueObjects: undefined,
+        };
+      }
+    } catch (e) {
+      // Not a date
+    }
+
+    try {
+      JSON.parse(data);
```

Comment:
this here is not good - please check for field type here as well - see what field types supposed to have json